src/Microsoft.FileFormats/MachO/MachOFile.cs

```diff
@@ -448,28 +448,93 @@ public class MachSymtab
+        private readonly Lazy<NList[]> _symbolTable;
 
         public MachSymtab(Reader machReader, ulong symtabPosition, ulong dysymtabPosition, Reader physicalAddressSpace)
         {
             _machReader = machReader;
             _physicalAddressSpace = physicalAddressSpace;
             _symtabLoadCommand = new Lazy<MachSymtabLoadCommand>(() => _machReader.Read<MachSymtabLoadCommand>(symtabPosition));
             _dysymtabLoadCommand = new Lazy<MachDySymtabLoadCommand>(() => _machReader.Read<MachDySymtabLoadCommand>(dysymtabPosition));
+            _stringReader = new Lazy<Reader>(GetStringReader);
+            _symbolTable = new Lazy<NList[]>(ReadSymbolTable);
             _symbols = new Lazy<MachSymbol[]>(ReadSymbols);
         }
 
         public IEnumerable<MachSymbol> Symbols { get { return _symbols.Value; } }
 
-        private MachSymbol[] ReadSymbols()
+        public bool TryLookupSymbol(string symbol, out ulong offset)
         {
+            if (symbol is null)
+                throw new ArgumentNullException(nameof(symbol));
+
             MachSymtabLoadCommand symtabLoadCommand = _symtabLoadCommand.Value;
             MachDySymtabLoadCommand dysymtabLoadCommand = _dysymtabLoadCommand.Value;
-            symtabLoadCommand.IsNSymsReasonable.CheckThrowing();
 
-            ulong position = symtabLoadCommand.SymOffset + (dysymtabLoadCommand.IExtDefSym * _physicalAddressSpace.SizeOf<NList>());
-            NList[] nlists = _physicalAddressSpace.ReadArray<NList>(position, dysymtabLoadCommand.NextDefSym);
-            Reader stringReader = _physicalAddressSpace.WithRelativeAddressSpace(symtabLoadCommand.StringOffset, symtabLoadCommand.StringSize);
-            return nlists.Select(n => new MachSymbol() { Name = stringReader.Read<string>(n.StringIndex), Raw = n }).ToArray();
+            // First, search just the "external" export symbols 
+            if (TryLookupSymbol(dysymtabLoadCommand.IExtDefSym, dysymtabLoadCommand.NextDefSym, symbol, out offset))
+            {
+                return true;
+            }
+
+            // If not found in external symbols, search all of them
+            if (TryLookupSymbol(0, symtabLoadCommand.SymCount, symbol, out offset))
+            {
+                return true;
+            }
+
+            offset = 0;
+            return false;
+        }
+
+        private bool TryLookupSymbol(uint start, uint nsyms, string symbol, out ulong offset)
+        {
+            NList[] symTable = _symbolTable.Value;
+            if (symTable is not null)
+            {
+                for (uint i = 0; i < nsyms && start + i < symTable.Length; i++)
+                {
+                    string name = _stringReader.Value.Read<string>(symTable[start + i].StringIndex); 
+                    if (name.Length > 0)
+                    {
+                        // Skip the leading underscores to match Linux externs
+                        if (name[0] == '_')
```

Comment:
Future todo: this is going to allocate for all linux externs for every module. If we see this being an issue, use System.Memory + Span.Slice + SequenceEquals.